### PR TITLE
ISPN-6623 Thread leak in the HotRod Server

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-6623

### CODE REVIEW

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/DistTopologyChangeUnderLoadSingleOwnerTest.java [hunk 1]
```diff
@@ -56,12 +56,14 @@ public void testRestartServerWhilePutting() throws Exception {
       // Start server
       EmbeddedCacheManager cm = TestCacheManagerFactory.createClusteredCacheManager(getCacheConfiguration());
       registerCacheManager(cm);
-      HotRodClientTestingUtil.startHotRodServer(cm);
+      HotRodServer server = HotRodClientTestingUtil.startHotRodServer(cm);
```
Comment (tristantarrant) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/DistTopologyChangeUnderLoadSingleOwnerTest.java]: Maybe wrap this with a try finally ?

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/DistTopologyChangeUnderLoadTest.java [hunk 2]
```diff
@@ -47,13 +48,14 @@ public void testPutsSucceedWhileTopologyChanges() throws Exception {
 
       EmbeddedCacheManager cm = TestCacheManagerFactory.createClusteredCacheManager(getCacheConfiguration());
       registerCacheManager(cm);
-      HotRodClientTestingUtil.startHotRodServer(cm);
+      HotRodServer newServer = HotRodClientTestingUtil.startHotRodServer(cm);
```
Comment (tristantarrant) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/DistTopologyChangeUnderLoadTest.java]: Here too (try finally)

**Comment by gustavonalle** [comment]
@tristantarrant updated

**Comment by wburns** [comment]
Changes seem fine to me, @tristantarrant ?

**Comment by tristantarrant** [comment]
Pushed to master.

**Pull request closed without merging**